jerry-core/debugger/jerry-debugger.c

```diff
@@ -1 +1,101 @@
-//TODO: Implement the socket communication
+/* Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerry-debugger.h"
+#include "jerry-port.h"
+
+#define PORT 5001
+#define BLACKLOG 1
+#define BUFFER_SIZE 128
+
+int sock;                     /**< return value of socket(), used other methods */
+int connected;                /**< return value of the whole socket connection */
+ssize_t byte_send;            /**< size of byte_send */
+size_t data_len;              /**< data lenght */
```

Comment:
Why is this global?

Moreover, you have a function parameter with the same name (but different type) and it is very misleading.